torch_xla/csrc/runtime/pjrt_computation_client.cc

```diff
@@ -990,13 +994,45 @@ ComputationClient::MemoryInfo PjRtComputationClient::GetMemoryInfo(
   };
 }
 
-const PJRT_Api* PjRtComputationClient::GetPjRtCApiIfAvailable() const {
-  // dynamic_cast will return a nullptr if the client is not PjRtCApiClient.
+void PjRtComputationClient::RegisterCustomCall(const std::string& fn_name,
+                                               void* function_ptr,
+                                               const std::string& platform) {
+  if (platform != "CUDA") {
+    XLA_ERROR() << "Custom call targets can only be registered for "
+                   "PJRT CUDA runtime.";
+    return;
+  }
+
   auto* c_api_client = dynamic_cast<xla::PjRtCApiClient*>(client_.get());
-  if (c_api_client) {
-    return c_api_client->pjrt_c_api();
+  if (!c_api_client) {
+    XLA_REGISTER_CUSTOM_CALL_TARGET_WITH_SYM(fn_name, function_ptr, platform);
+    return;
+  }
+  const PJRT_Api* pjrt_api = c_api_client->pjrt_c_api();
+
+  // See openxla reference:
+  // https://github.com/openxla/xla/blob/b604c8d87df842002a7a8de79a434026329fbcb2/xla/pjrt/c/pjrt_c_api_gpu_test.cc#L414
+  const PJRT_Extension_Base* next =
+      reinterpret_cast<const PJRT_Extension_Base*>(pjrt_api->extension_start);
+  while (next != nullptr &&
+         next->type !=
+             PJRT_Extension_Type::PJRT_Extension_Type_Gpu_Custom_Call) {
+    next = next->next;
+  }
+  if (next == nullptr) {
+    return;
```

Comment:
shouldn't we throw a warning message or something for registration failure?

Reply:
Yeah, you're right. I meant to go back to this and forgot.